docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Text/FeaturizeTextWithOptions.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+using Microsoft.ML.Transforms.Text;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public static class FeaturizeTextWithOptions
+    {
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Get a small dataset as an IEnumerable.
+            var samples = new List<TextData>()
+            {
+                new TextData(){ Text ="ML.NET's FeaturizeText API uses a composition of several basic transforms to convert text into numeric features." },
+                new TextData(){ Text ="This API can be used as a featurizer to perform text classification." },
+                new TextData(){ Text ="There are a number of approaches to text classification." },
+                new TextData(){ Text ="One of the simplest and most common approaches is called “Bag of Words”." },
+                new TextData(){ Text ="Text classification can be used for a wide variety of tasks" },
+                new TextData(){ Text ="such as sentiment analysis, topic detection, intent identification etc." },
+            };
+
+            // Convert training data to IDataView.
+            var dataview = mlContext.Data.LoadFromEnumerable(samples);
+
+            // A pipeline for converting text into numeric features.
+            // The following call to 'FeaturizeText' instantiates 'TextFeaturizingEstimator' with given parameters.
+            // The length of the output feature vector depends on these settings.
+            var options = new TextFeaturizingEstimator.Options()
+            {
+                // Also output tokenized words
+                OutputTokensColumnName = "OutputTokens",
+                CaseMode = TextNormalizingEstimator.CaseMode.Lower,
+                // Use ML.NET's built-in stop word remover
+                StopWordsRemoverOptions = new StopWordsRemovingEstimator.Options() { Language = TextFeaturizingEstimator.Language.English },
+                WordFeatureExtractor = new WordBagEstimator.Options() { NgramLength = 1 },
+                CharFeatureExtractor = new WordBagEstimator.Options() { NgramLength = 1 },
```

Comment:
>1 [](start = 86, length = 1)

is this single char tokenization? it would be just the alphabets. is it ever useful?

Reply:
Yes, can be useful in some cases. but lets change it to 3-gram which is more useful.

---
In reply to: [269776047](https://github.com/dotnet/machinelearning/pull/3120#discussion_r269776047) [](ancestors = 269776047)